src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -354,6 +418,66 @@ private async Task<IAnonymizer> CreateAnonymizerAsync(CancellationToken cancella
                 queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.Type, _exportJobRecord.ResourceType));
             }
 
+            var filteredResources = new HashSet<string>();
+            if (_exportJobRecord.Filters != null)
+            {
+                foreach (var filter in _exportJobRecord.Filters)
+                {
+                    filteredResources.Add(filter.ResourceType);
+                }
+            }
+
+            if (!progress.RunFilteredSearches)
+            {
+                var index = -1;
+                if (_exportJobRecord.Filters != null)
+                {
+                    foreach (var filter in _exportJobRecord.Filters)
+                    {
+                        index++;
+                        if (progress.CurrentFilter != null && filter != progress.CurrentFilter)
+                        {
+                            continue;
+                        }
+
+                        progress.SetFilter(filter);
+
+                        List<Tuple<string, string>> filterQueryParametersList = new List<Tuple<string, string>>(queryParametersList);
+                        filterQueryParametersList.Add(Tuple.Create(KnownQueryParameterNames.Type, filter.ResourceType));
```

Comment:
We are adding the resourceType to the query parameters here but also passing the filter.ResourceType to search below (where it gets added as a "higher" level filter). Seems like we can remove one of them.